geonode/harvesting/harvesters/wms.py

```diff
@@ -104,66 +136,179 @@ def check_availability(self, timeout_seconds: typing.Optional[int] = 5) -> bool:
+                            license=[],
+                            abstract=layer['abstract'],
+                            spatial_extent=layer['spatial_extent']
+                        ),
+                        distribution=resourcedescriptor.RecordDistribution(
+                            legend_url=layer['legend_url'],
+                            wms_url=layer['wms_url']
+                        ),
+                        reference_systems=layer['crs'],
+                    ),
+                    additional_information=None
+                )
+        return None
+
+    def _get_data(self) -> dict:
+        """ Return data from the harvester url in json format"""
+        params = self._base_wms_parameters.copy()
         params.update({
             "request": "GetCapabilities",
         })
         get_capabilities_response = self.http_session.get(
             self.remote_url, params=params)
         get_capabilities_response.raise_for_status()
+
         root = etree.fromstring(get_capabilities_response.content, parser=XML_PARSER)
         nsmap = _get_nsmap(root.nsmap)
-        useful_layers_elements = []
+
+        # get layers from element tree
+        layers = []
         leaf_layers = root.xpath("//wms:Layer[not(.//wms:Layer)]", namespaces=nsmap)
         for layer_element in leaf_layers:
-            try:
-                title = layer_element.xpath("wms:Title/text()", namespaces=nsmap)[0]
-            except IndexError:
-                name = layer_element.xpath("wms:Name/text()", namespaces=nsmap)[0]
-                title = name
+            data = self._layer_element_to_json(layer_element)
+            title = data['title']
             if self.layer_title_filter is not None:
                 if self.layer_title_filter.lower() not in title.lower():
                     continue
-            logger.debug(f"Creating resource descriptor for layer {title!r}...")
-        self.update_harvesting_session(
-            harvesting_session_id, total_records_found=len(useful_layers_elements))
-        self.finish_harvesting_session(harvesting_session_id)
+            layers.append(data)
+        return {
+            'contact': self._get_contact(root),
+            'layers': layers
+        }
 
-    def update_geonode_resource(
-            self,
-            harvested_info: base.HarvestedResourceInfo,
-            harvestable_resource: "HarvestableResource",  # noqa
-            harvesting_session_id: int,
-    ):
-        raise NotImplementedError
+    def _get_contact(self, element: etree.Element) -> dict:
+        """Return contact from element"""
+        nsmap = _get_nsmap(
+            element.nsmap)
+        service = get_xpath_elements(element, "wms:Service", nsmap)[0]
+        contact = get_xpath_elements(service, "wms:ContactInformation", nsmap)[0]
+        contact_person = get_xpath_elements(contact, "wms:ContactPersonPrimary", nsmap)[0]
+        address = get_xpath_elements(contact, "wms:ContactAddress", nsmap)[0]
```

Comment:
There doesn't seem to be a need to use `get_xpath_elements()` here - the simpler `element.xpath()` provided by `etree` would suffice, _i.e._:

```suggestion
        contact = element.xpath("wms:Service/wms:ContactInformation", nsmap)[0]
        contact_person = contact.xpath("wms:ContactPersonPrimary", nsmap)[0]
        address = contact.xpath("wms:ContactAddress", nsmap)[0]
```